department-of-veterans-affairs/caseflow
# APPEALS-13212: Email templates

Resolves [APPEALS-13212](https://vajira.max.gov/browse/APPEALS-13212)

### Description
Create Email templates for the case flow mailers.

### Acceptance Criteria
- [x] Creation of a base mail template that can be reused for each of the respective scenarios (i.e. VA header, Help Desk message and signature)
- [x] Creation of each of the nine individual templates which will be based off the template created in the first AC

### Testing Plan
This can be tested from rails console. It needs some configuration on local machine for this to be testable. Here are the steps.

1. create a file called `membership_request_mailer.rb` under `app/mailers/`. Use the following content for it
 ```
class MembershipRequestMailer < ActionMailer::Base
  default from: "Board of Veterans' Appeals <BoardofVeteransAppealsHearings@messages.va.gov>"
  layout "membership_request_mailer"

  def request_received
    @user = User.first
    @requesting_org_names = ['Organization One', 'Organization Two']
    mail(to: "template_testing@mailinator.com", subject: "Testing mailer template")
  end
end
```

Note: Remember to update the method name if you are testing this for multiple templates. In this case we are using `request_received` template, thats why we have the method name as `request_received`

2. Now, go to rails console `rails console` from terminal and paste the following command
```
print MembershipRequestMailer.request_received.body
```

3. Copy the content from `<!DOCTYPE html>` until `</html>`. 
4. Open an empty file and paste the content you copied from number 3. Save it as `request_received.html`
5. Open the html, you should see the formatted code on the browser.

Video link

https://user-images.githubusercontent.com/27713962/218568921-406d4749-d9bf-4e6c-82d2-39f8db7e8d65.mp4
